### PR TITLE
Start menu suggestions: announce result details

### LINKED ISSUE #10329: Windows 10 start menu: search results are not announced with details with recent feature updates
Hi,

Initially, this was a Version 1903 change, but older feature updates do exhibit this problem. A fix is now part of Windows 10 App Essentials add-on and will be part of a separate PR from #10328.

### Background:
In Version 1903 (May 2019 Update), Start menu search experience received an overhaul; powered by window2s Search in addition to being separated somewhat from Cortana (the latter is now the case with 20H1 builds). As part of this redesign (which was subsequently backported to older feature updates via server-side change), results are not announced with details as it used to. Turns out UIA tree is different - sort of a treeview, with categories as results and underneath them are actual results, thus when announcing results, grandparent (not the parent) of the entry must be consulted (thankfully, the parent is recognized as a suggestion list item).

### Steps to reproduce:
For this, NVDA must run with add-ons disabled:

1. Open Start menu.
2. Search for things.
3. Use up or down arrow keys to review search results.

### Actual behavior:
Only the search box content is announced when reviewing results.

### Expected behavior:
Details such as category and position information is announced.

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
2019.3 alpha.18874

#### Windows version:
Windows 10 Version 1909 (build 18363)

#### Name and version of other software in use when reproducing the issue:
None

#### Other information about your system:
None

#### Does the issue still occur after restarting your PC?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes - older versions such as 2018.4 and 21019.2.

### Technical:
See the background section for details. In short, when it comes to recognizing suggestion list item, look at obj.parent instance.

Thanks.

### PR DESCRIPTION
**Pull request by josephsl**
Hi,

First of a series of pull requests on updating Start menu support routines:

### Link to issue number:
Fixes #10329 

### Summary of the issue:
Details for search results are not announced like in the past.

### Description of how this pull request fixes the issue:
Due to restructured UIA tree for Start menu (applicable in Version 1607/Anniversary Update with updated Windows Search design), search results are housed inside search categories, with categories themselves reported as suggestions list items. Therefore recognize result entries as seen in newer Windows Search experience.

### Testing performed:
Tested via Windows 10 App Essentials on Version 1607 and later (no need to look into 1507 and 1511 because they retain the original Windows 10 Start menu experience).

### Known issues with pull request:
Results will be announced twice due to caret movement script (see #10336 and a follow-up PR for a fix).

### Change log entry:
Bug fixes:

In Start menu for Windows 10 Anniversary Update and later, NVDA will announce details of search results. (#10232

### CODE REVIEW

**Comment by michaelDCurran** [comment]
In 18995, It looks as though the suggestions in each category are housed in an unlabeled list item which is a sibling to the category listItem itself. I.e. If I type "word" There is a "find results in apps" listItem, and its next sibling is an unlabeled listItem, which contains "Microsoft Word Mobile" "Microsoft Word 2007" etc.
Neither these suggestions nor the unlabelled listItem have a useful automationID or className. 
Is this also what you are seeing?
In short, this pr does not seem to make anydifference for me on this build of Windows.

**Comment by josephsl** [comment]
Hi. Exactly the same, even in 18362 and 18363. The “cool” thing is that this unlabeled item is seen as suggestion list item when in fact results are inside them, akin to a treeview structure. Thanks.

File: source/appModules/searchui.py [hunk 1]
```diff
@@ -25,3 +29,7 @@ def chooseNVDAObjectOverlayClasses(self,obj,clsList):
 		elif isinstance(obj, UIA):
 			if obj.UIAElement.cachedAutomationID == "SearchTextBox":
 				clsList.insert(0, StartMenuSearchField)
+			# #10329: Since 2019, some suggestion items are grouped inside another suggestions list item.
+			# Because of this, result details will not be announced like in the past.
+			elif obj.role == controlTypes.ROLE_LISTITEM and isinstance(obj.parent, SuggestionListItem):
```
Comment (leonardder) [inline source/appModules/searchui.py]: Wouldn't it be possible to add this extra check to the `findOverlayClasses` function in NVDAObjects.UIA? I have the feelling that current code scatters the places where `SuggestionListItem` is added as an overlay.

**Comment by josephsl** [comment]
Hi,

Ah, I see what's up: executable has been renamed, that's why you are not seeing changes. I'll file a PR that'll support this PR and a host of Windows Search PR's (I should have done that before doing this PR series)...

Thanks.

**Comment by josephsl** [comment]
Hi, in a way, that would be ideal, but I think sometimes it is helpful to let app modules recognize objects like this if it is a custom implementation or does not follow specs correctly. I learned it the hard way this year when dealing with search field in Edge – we were catching all sorts of search fields when in fact Edge should have been excluded in the first place. If you believe UIA.findOverlayClasses is the way to go (and if others agree), I’m willing to make that change tonight. Thanks.

**Comment by leonardder** [comment]
I don't insist on having it moved to UIA.findOverlayClasses. However, I 
think it must be documented somehow what suggestion items are covered by 
UIA.findOverlayClasses and what suggestion items are not.

**Comment by josephsl** [comment]
Hi, I think it can be done as part of SuggestionsListItem class docstring, as well as a comment when introducing this class in find overlay classes method. Thanks for pointing this out (shall be done tonight).

**Comment by josephsl** [comment]
Hi,

To @MichaelDCurran - can you try testing the PR with latest commit which adds searchapp app module for 20H1?

Thanks.

**Comment by michaelDCurran** [comment]
@josephsl  Yes, this is now working great on 18995 :)

**Review by michaelDCurran — APPROVED** [review]
